dotcom-rendering/src/web/server/extractIslands.ts

```diff
@@ -0,0 +1,21 @@
+const regex = new RegExp(
+	// Gets the name of any island where expediteLoading is set to 'true'
+	// e.g <gu-island name="MyComponent" props="..." expediteLoading="true"></gu-island>
+	/<gu-island name="([a-zA-Z]+)"[^>]*expediteLoading="true"[^>]*><\/gu-island>/g,
+);
+
+// https://stackoverflow.com/a/9229821/17210412
+const unique = (arr: string[]): string[] => {
+	return [...new Set(arr)];
+};
+
+export const extractExpeditedIslands = (html: string): string[] => {
+	const matches = html.matchAll(regex);
+
+	return unique(
+		Array.from(matches).map((match) => {
+			// MyComponent.importable becomes MyComponent-importable during the build
+			return `${match[1]}-importable`;
+		}),
+	);
```

Comment:
Why the `unique` indirection? I would prefer to spread that `Set` directly.

Reply:
I don't really mind! It was a little hard to parse with so much happening at once for me:
```js
[...new Set(Array.from(matches).map((match) => {
	// MyComponent.importable becomes MyComponent-importable during the build
	return `${match[1]}-importable`;
}),
```
Calling it `unique` or `deduplicate` makes it more clear what the purpose is

Reply:
This can happen in a subsequent step. And maybe pick spreading or `Array.from`, but not both?

```ts
const islands = [...matches].map(([,match]) => 
	// MyComponent.importable becomes MyComponent-importable during the build
	`${match}-importable`
);
return [...new Set(islands)];
```

Reply:
Much tidier! Thank you :)